### PR TITLE
SINGA-500 add onnx example models

### PR DESCRIPTION
**Pull request by joddiy**

### CODE REVIEW

**Comment by joddiy** [comment]
@nudles ready to merge

File: python/singa/utils.py [hunk 1]
```diff
@@ -231,3 +234,40 @@ def force_unicode(s):
         return s.decode('utf-8')
     except AttributeError:
         return s
+
+
+def postorderRecursive(root, root_t):
```
Comment (nudles) [inline python/singa/utils.py]: post_order_recursive? 
be consistent in naming.

File: examples/onnx/arcface.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import os
```
Comment (nudles) [inline examples/onnx/arcface.py]: need to add license boilerplate.

File: examples/onnx/bert/bert-squad.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+class Infer:
+
+    def __init__(self, sg_ir):
+        self.sg_ir = sg_ir
+
+    def forward(self, x):
+        return sg_ir.run(x)
+
+
+def preprocessing():
```
Comment (nudles) [inline examples/onnx/bert/bert-squad.py]: preprocess? function name can be verb.

File: examples/onnx/bert/bert-squad.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+    eval_examples = read_squad_examples(input_file=predict_file)
+
+    # Use convert_examples_to_features method from run_onnx_squad to get parameters from the input
+    input_ids, input_mask, segment_ids, extra_data = convert_examples_to_features(
+        eval_examples, tokenizer, max_seq_length, doc_stride, max_query_length)
+    return input_ids, input_mask, segment_ids, extra_data, eval_examples
+
+
+def postprocessing(eval_examples, extra_data, all_results):
```
Comment (nudles) [inline examples/onnx/bert/bert-squad.py]: postprocess?

File: examples/onnx/bert/run_onnx_squad.py [hunk 5]
```diff
@@ -0,0 +1,507 @@
+# Copyright 2018 The Google AI Language Team Authors.
```
Comment (nudles) [inline examples/onnx/bert/run_onnx_squad.py]: need to add this license into the LICENSE file.

**Comment by joddiy** [comment]
> Thanks, @joddiy !
> I have added some minor comments.

Done, thanks for reviewing. Please wait for a while, I will check the Matmul with shicong again. After that, we can merge this PR.

**Comment by joddiy** [comment]
@nudles ready to merge, please merge #639 firstly.

**Comment by nudles** [comment]
Is there any dependency on issue #635  and #634 ?

**Comment by joddiy** [comment]
> Is there any dependency on issue #635 and #634 ?

#634 has been fixed, I guess you mean #633 and #635, actually, I indeed need shicong to fix them, however, I modify some code within autograd.py, it can run now, although, not so elegant...

maybe later, when shicong has fixed these two issues, I will update the autogard.py then. But for now, the current implementation can still work.

**Pull request merged**